adserver/views.py

```diff
@@ -221,16 +223,14 @@ def get_context_data(self, **kwargs):
 
     def is_advertiser_new(self):
         """Advertisers are new if there's never been an ad impression for them."""
-        return not AdImpression.objects.filter(
-            advertisement__flight__campaign__advertiser_id=self.advertiser.id
+        return not AdvertiserImpression.objects.filter(
+            advertiser_id=self.advertiser.id
         ).exists()
 
     def has_views_this_month(self, start_date):
         """Detect if advertisers have impressions since the start date (first of the month)."""
         return (
-            AdImpression.objects.filter(
-                advertisement__flight__campaign__advertiser_id=self.advertiser.id
-            )
+            AdvertiserImpression.objects.filter(advertiser_id=self.advertiser.id)
```

Comment:
Seems this is snuck in with a Publisher PR, but good to get them both done at once :)